fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/RuleSet.java

```diff
@@ -274,8 +274,8 @@ public class RuleSet {
         ));
 
         rewriteRules.put(RuleSetType.MULTI_DISTINCT_REWRITE, ImmutableList.of(
-                new RewriteMultiDistinctRule(),
-                new RewriteMultiDistinctByCTERule()
+                new RewriteMultiDistinctByCTERule(),
```

Comment:
Why change the order?

Reply:
rewrit to cte first, and rewrite other to multi distinct if cte not support, don't depend on the cte flag